### PR TITLE
Compare all the things.

### PR DESCRIPTION
**Pull request by miscco**
# Description
This addresses LWG-3330 that requires a bunch of headers to include `<compare>`
Luckily nearly everything includes `<xutility>` so we more or less only need to add it to `<utility>` and two lone other headers

# Checklist

Be sure you've read README.md and understand the scope of this repo.

If you're unsure about a box, leave it unchecked. A maintainer will help you.

- [x] Identifiers in product code changes are properly `_Ugly` as per
  https://eel.is/c++draft/lex.name#3.1 or there are no product code changes.
- [ ] The STL builds successfully and all tests have passed (must be manually
  verified by an STL maintainer before automated testing is enabled on GitHub,
  leave this unchecked for initial submission).
- [x] These changes introduce no known ABI breaks (adding members, renaming
  members, adding virtual functions, changing whether a type is an aggregate
  or trivially copyable, etc.).
- [x] These changes were written from scratch using only this repository,
  the C++ Working Draft (including any cited standards), other WG21 papers
  (excluding reference implementations outside of proposed standard wording),
  and LWG issues as reference material. If they were derived from a project
  that's already listed in NOTICE.txt, that's fine, but please mention it.
  If they were derived from any other project (including Boost and libc++,
  which are not yet listed in NOTICE.txt), you *must* mention it here,
  so we can determine whether the license is compatible and what else needs
  to be done.

### CODE REVIEW

**Comment by StephanTLavavej** [comment]
We should add test coverage to verify that all of these inclusions aren't affected by future header refactorings. A single test directory with multiple translation units, each with `#include <meow>` and attempting to name something from `<compare>`, should be sufficient.

File: stl/inc/system_error [hunk 1]
```diff
@@ -15,6 +15,10 @@
 #include <xcall_once.h>
 #include <xerrc.h>
 
+#if _HAS_CXX20 && defined(__cpp_impl_three_way_comparison)
+#include <compare>
+#endif // _HAS_CXX20 && defined(__cpp_impl_three_way_comparison)
+
```
Comment (StephanTLavavej) [inline stl/inc/system_error]: This is unnecessary:
https://github.com/microsoft/STL/blob/d4ee5c3fb4abfe8492f2108384d6f4e0fd0afeae/stl/inc/system_error#L14
https://github.com/microsoft/STL/blob/d4ee5c3fb4abfe8492f2108384d6f4e0fd0afeae/stl/inc/stdexcept#L12
https://github.com/microsoft/STL/blob/d4ee5c3fb4abfe8492f2108384d6f4e0fd0afeae/stl/inc/xstring#L14
https://github.com/microsoft/STL/blob/d4ee5c3fb4abfe8492f2108384d6f4e0fd0afeae/stl/inc/xmemory#L16
https://github.com/microsoft/STL/blob/d4ee5c3fb4abfe8492f2108384d6f4e0fd0afeae/stl/inc/xutility#L15

**Comment by StephanTLavavej** [comment]
(It appears that creating new files and saving them as `.cpp` might not activate format-on-save; I'll have to remember that in the future.)

**Comment by StephanTLavavej** [comment]
Update: This is blocked by an Internal Compiler Error in our modules integration tests, which we need to reduce and report to the compiler team.

**Comment by StephanTLavavej** [comment]
Reported DevCom-937601 reduced to a library-free test case.

File: tests/std/tests/GH_000545_include_compare/test_ranges.cpp [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <version> // TRANSITION, EDG support for concepts
```
Comment (CaseyCarter) [inline tests/std/tests/GH_000545_include_compare/test_ranges.cpp]: I'd prefer this to be `// TRANSITION, GH-395`, but it's probably not worth the trouble to make just that change now.

**Comment by StephanTLavavej** [comment]
The compiler has been fixed thanks to Cameron DaCamara, so I've merged this. Thanks again for implementing this LWG issue and unblocking further work!

**Pull request merged**